packages/data/src/controls.js

```diff
@@ -30,8 +37,15 @@ const DISPATCH = '@@data/DISPATCH';
  *
  * @return {Object} The control descriptor.
  */
-function select( storeKey, selectorName, ...args ) {
-	return { type: SELECT, storeKey, selectorName, args };
+function select( storeNameOrDefinition, selectorName, ...args ) {
+	return {
+		type: SELECT,
+		storeKey: isObject( storeNameOrDefinition )
```

Comment:
Potentially this check could be moved to the implementation of the control itself. Actually, I wonder if it's needed at all since this should just call `registry.select` which handles both already right? (same for the others)

Reply:
You are probably correct. There are some levels of indirection here so I would have to do a second pass of refactorings. I will merge this PR and open another one. I suspect that some unit tests will have to be revised so they work with `storeKey` as an object. I also wasn't sure if I can update `storeKey` key.